app/models/ability.rb

```diff
@@ -2,19 +2,78 @@ class Ability
   include Hydra::Ability
   include Sufia::Ability
 
-  
   # Define any customized permissions here.
   def custom_permissions
-    # Limits deleting objects to a the admin user
-    #
-    # if current_user.admin?
-    #   can [:destroy], ActiveFedora::Base
-    # end
-
-    # Limits creating new objects to a specific group
-    #
-    # if user_groups.include? 'special_group'
-    #   can [:create], ActiveFedora::Base
-    # end
+    discover_permissions
+    override_download_permissions
+  end
+
+  private
+
+  # TODO: discover_permissions and friends should really be moved into
+  # hydra_access_controls.  We should fork that gem and make the change
+  # there instead, then submit it as a PR.
+  #
+  # override_download_permissions should stay here, but maybe there's a
+  # way to not duplicate the code from hydra_access_controls.
+
+  def discover_permissions
+    can :discover, String do |id|
+      test_discover(id)
+    end
+
+    can :discover, ActiveFedora::Base do |obj|
+      test_discover(obj.id)
+    end
+
+    can :discover, SolrDocument do |obj|
+      cache.put(obj.id, obj)
+      test_discover(obj.id)
+    end
+  end
+
+  def override_download_permissions
+    can :download, ActiveFedora::File do |file|
+      parent_uri = file.uri.to_s.sub(/\/[^\/]*$/, '')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.